tests/components/alexa/test_entities.py

```diff
@@ -16,3 +18,29 @@ async def test_unsupported_domain(hass):
     msg = msg["event"]
 
     assert not msg["payload"]["endpoints"]
+
+
+async def test_serialize_discovery_recovers(hass, caplog):
+    """Test we handle an interface raising unexpectedly during serialize discovery."""
+    request = get_new_request("Alexa.Discovery", "Discover")
+
+    hass.states.async_set("switch.bla", "on", {"friendly_name": "Boop Woz"})
+
+    with patch(
+        "homeassistant.components.alexa.capabilities.AlexaPowerController.serialize_discovery",
+        side_effect=TypeError,
+    ):
+        msg = await smart_home.async_handle_message(hass, DEFAULT_CONFIG, request)
+
+    assert "event" in msg
+    msg = msg["event"]
+
+    interfaces = {
+        ifc["interface"] for ifc in msg["payload"]["endpoints"][0]["capabilities"]
+    }
+
+    assert "Alexa.PowerController" not in interfaces
+    assert (
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)